generative_ai/function_calling/test_function_calling.py

```diff
@@ -85,3 +92,27 @@ def test_function_calling_chat() -> None:
 def test_parallel_function_calling() -> None:
     response = parallel_function_calling_example.parallel_function_calling_example()
     assert response is not None
+
+
+def test_function_calling_app() -> None:
+    result = function_calling_application.create_app()
+    assert result["weather_response"] is not None
+
+    tool = result["tool"]
+    model = GenerativeModel(model_name="gemini-1.5-pro-002", tools=[tool])
+    chat_session = model.start_chat()
+
+    response = chat_session.send_message("What will be 1 multiplied by 2?")
+    assert response is not None
+
+    extract_sales_prompt = """
```

Comment:
I am adding a test for each toolbox function created by this sample because of [this reason](https://github.com/GoogleCloudPlatform/python-docs-samples/pull/12811#discussion_r1866424340). The toolbox is lazily evaluated, and it's not enough to simply create a function declaration to check that it's working. 
Only when we trigger a "function call" response from the model with a target prompt it validates the declaration and can trigger an error.